pkg/didcomm/common/service/service.go

```diff
@@ -30,6 +45,33 @@ type DIDCommMsg struct {
 	Payload  []byte
 	// TODO : might need refactor as per the issue-226
 	OutboundDestination *Destination
+	Common              *Common
+}
+
+// ThreadID returns msg ~thread.thid if there is no ~thread.thid returns msg @id
+// message is invalid if ~thread.thid exist and @id is absent
+func (m *DIDCommMsg) ThreadID() (string, error) {
+	if m.Common == nil {
```

Comment:
This logic should be removed.

Reply:
refactored! but this logic still there, check it out